notebook2/notebook/notebook.py

```diff
@@ -188,6 +198,131 @@ def external_url_for(path):
     return url + path
 
 
+# Kube has max 63 character limit
+def user_id_transform(user_id): return hashlib.sha224(user_id.encode('utf-8')).hexdigest()
+
+
+def read_svc_status(svc_name: str):
+    try:
+        # TODO: inspect exception for non-404
+        _ = k8s.read_namespaced_service(svc_name, 'default')
+        return 'Running'
+    except Exception:
+        return 'Deleted'
+
+
+def read_containers_status(container_statuses):
```

Comment:
this method seems unnecessary. It is almost exactly 
```python
return [x.to_dict() for x in container_statuses]
```

Man it is infuriating that k8s uses `None` instead of `[]` for empty lists. I think we can change line 289 this and completely remove this method.
```python
    container_statuses = pod.status.container_statuses
    # ...
            'container_status': container_statuses or [x.to_dict() for x in container_statuses]
```

Reply:
Sounds good

Reply:
So the method exists because I want to return only a part of the full response. A public interface for the web app, regardless of the representation returned by K1.